src/_pytest/python.py

```diff
@@ -151,15 +152,16 @@ def pytest_configure(config):
 @hookimpl(trylast=True)
 def pytest_pyfunc_call(pyfuncitem):
     testfunction = pyfuncitem.obj
-    iscoroutinefunction = getattr(inspect, "iscoroutinefunction", None)
-    if iscoroutinefunction is not None and iscoroutinefunction(testfunction):
-        msg = "Coroutine functions are not natively supported and have been skipped.\n"
+    if iscoroutinefunction(testfunction) or (
+        sys.version_info >= (3, 6) and inspect.isasyncgenfunction(testfunction)
```

Comment:
What do you think about having a `isasyncgenfunction` in `_pytest.compat` as well, and just returning `False` for Python 3.5, since it can't define these?

Reply:
I actually disagree, we only use it here and when support for 3.5 is dropped isasyncgenfunction in compat will look very strange